snuba_sdk/expressions.py

```diff
@@ -66,7 +68,13 @@ def is_scalar(value: Any) -> bool:
     return False
 
 
-column_name_re = re.compile(r"^[a-zA-Z][a-zA-Z_.]+$")
+column_name_re = re.compile(r"^[a-zA-Z](\w|\.)+$")
+# In theory the function matcher should be the same as the column one.
+# However legacy API sends curried functions as raw strings, and it
+# wasn't worth it to import an entire parsing grammar into the SDK
+# just to accomodate that one case. Instead, allow it for now and
+# once that use case is eliminated we can remove this.
+function_name_re = re.compile(r"^[a-zA-Z](\w|[().,]| |\[|\])+$")
```

Comment:
What would the incentive to remove this use case be ? If the clients will find it convenient they will not stop using it.

Reply:
Well right now the clients have to use it in order to do things like parametric functions. Once we have SnQL everywhere, they no longer have to use it and we can deprecate it completely.